### DISCUSSION TITLE
Can I append a column on search results?

### DISCUSSION THREAD

**Discussion by nmfrankel** [prisma/prisma#15608] [Q&A]
## Problem

I'm creating a chat app and would like to return chat threads and "create" a column that returns the other users id.
Is there a way to do that with the Prisma client?

Here's a `SQL` query for greater detail, the `partner` column is what I would like to recreate.

```sql
SELECT
    IF(`senderId` = '1', `receiverId`, `senderId`) AS partner,
    `content` AS 'latest_msg',
    ...
FROM `msgs`
WHERE `senderId` = '1' OR `receiverId` = '1'
GROUP BY `partner`
```

**Comment by austincrim** [comment]
Hey @nmfrankel 👋,

This is not possible today without using [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw), but it should be unlocked when [Prisma Client Extensions](https://github.com/prisma/prisma/issues/15074) are released.

It would be awesome if you could take a look at the design and provide any feedback you have!

**Reply by nmfrankel** [reply]
Thank you for reminding me about that thread. On it I asked on it a while back whether this case would be possible with the new feature.

Additionally, from the thread I wasn't sure if I would be able to filter results `where: { ... }` based on the computed fields.

Would the proposal offer the option to create nested queries like the following one.

```sql
SELECT
	*,
	(
		SELECT a.metadata FROM messages a
		WHERE a.receiverId = '1' AND a.senderId = partner
		ORDER BY time DESC LIMIT 1
	) AS read_receipts
FROM (
	SELECT
		id,
		IF(senderId = 1, receiverId, senderId) AS partner,
		content,
		time
	FROM messages
	WHERE senderId = '1' OR receiverId = '1'
	ORDER BY time DESC
) AS _
GROUP BY partner;
```

Alt, would this simpiler query work?

```sql
SELECT
	id,
	IF(senderId = 1, receiverId, senderId) AS partner,
	content,
	MIN(metadata),
	time
FROM `messages`
WHERE senderId = '1' OR receiverId = '1'
GROUP BY partner
ORDER BY id DESC, time DESC
```

**Reply by austincrim** [reply]
Prisma Client Extensions would allow you to centralize and abstract the [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) call behind a named method that can be shared, which I think would be an improvement in this case.

**Reply by vladi-stevanovic** [reply]
👋 Hello @nmfrankel did you perhaps have a chance to check Austin's recommendation? Let us know if you still have any questions, or feel free to "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
